codegen-core/src/main/kotlin/software/amazon/smithy/rust/codegen/core/smithy/generators/UnionGenerator.kt

```diff
@@ -114,7 +116,44 @@ class UnionGenerator(
         const val UnknownVariantName = "Unknown"
     }
 }
+
 fun unknownVariantError(union: String) =
     "Cannot serialize `$union::${UnionGenerator.UnknownVariantName}` for the request. " +
         "The `Unknown` variant is intended for responses only. " +
         "It occurs when an outdated client is used after a new enum variant was added on the server side."
+
+private fun RustWriter.renderVariant(symbolProvider: SymbolProvider, member: MemberShape, memberSymbol: Symbol) {
+    if (member.target.name == "Unit") {
```

Comment:
The full ID should be checked just in case the model declares its own `Unit` shape. It will be [`smithy.api#Unit`](https://smithy.io/2.0/spec/model.html#unit-type).

Reply:
Thanks for point it out! Addressed in 2b148aa, but the way it's done is not ideal
- creating `ShapeId.from("smithy.api#Unit")` every time a comparison is performed
- multiple places need to be aware of special casing

Planning to refactor these in a subsequent commit.

Reply:
The first bullet was taken care of by fd60317.

The second bullet is not so straightforward. I haven't uncovered a reasonable abstraction that covers all the places in question; each place has a slightly different logic in its if-else expression. For instance, some call their file-local functions in the `else` block and those functions cannot be moved over to a refactored function which might be in a different file; callback lambdas are not that effective for abstraction here either because each call site still has to specify `Writable` for the `if` case and and the `else` case, making it aware that there is an if-else bifurcation based on whether a `Union` member is of the `Unit` type.

Reply:
For now, we'll keep what we have to keep it simple.